### PR TITLE
Implement code logic for registering a collision

### LINKED ISSUE #49: Register collision
The car will be able to register when it's in a collision

## Acceptance Criteria:

- [x] The car will registered been in a collision under the following criteria.
	- [x] The car will register a collision if the angle of change is above  55 or last -55
	- [x] when the car is stationary and and the car moved even though it did not get input from the user it will register a collision if the angle is increased more than two
- [x] the car will register a collision if one of the ultra sensors registers a distance of 23 or under

### PR DESCRIPTION
**Pull request by liamaxelrod**
RegisterCrash issue 49 criteria have been met.

The car will register a crash if the ultra sensors come to close to the car. The car will also register if the angle is changed more than 55 above or below because it is not possible for the car to do that naturally. Finally if the car is stationary or moving at 0.000400 and the angle changes more than 2° the car will register crash.

I want to merge this into the master.

close #49 

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -149,15 +275,18 @@ void loop()
...
+            if (!activeAvoidance){
+                stopZoneAutoBreak(frontUltDis, backUltDis);  
             }
-            incomingAvoidanceThreshold(frontUltDis, backIRDis);
+            incomingAvoidanceThreshold(frontUltDis, backUltDis);
+            
+        } else {
+                driveForwards = true;
+                drivebackwards = true;
         }
```
Comment (liamaxelrod) [inline arduino/smartcar/smartcar.ino]: I don't know why it's in the master version and not in mine is this a problem

Comment (liamaxelrod) [reply arduino/smartcar/smartcar.ino]: #ifdef __SMCE__	#ifdef __SMCE__
@@ -173,95 +302,140 @@ void loop()

**Comment by liamaxelrod** [comment]
I went through all the conflicts that were found when the merge was pulled to the master so when one of you can please review the code

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (liamaxelrod) [reply arduino/smartcar/smartcar.ino]: I made this comment looking at something different so you don't have to worry about this comment when it comes to terms this pull request

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -126,33 +147,24 @@ void setup()
...
-//         if (drivebackwards && speedGate <= 0){
-//             theSpeed = speedGate;
-//         }
-//             //Serial.println(theSpeed);//shows the speed
-        
+    {
         if (topic == "/smartcar/control/throttle") {
-            Serial.println(message);
-            car.setSpeed(message.toInt());//this needs to be swapped if the above is uncommented <<car.setSpeed(theSpeed);>>
+            Serial.println(message);//Liam are these needed
```
Comment (Lindet94) [inline arduino/smartcar/smartcar.ino]: These comments here. Can the comments or the code be removed?

File: arduino/smartcar/smartcar.ino [hunk 3]
```diff
@@ -126,33 +147,24 @@ void setup()
...
+            car.setSpeed(message.toInt());
         } else if (topic == "/smartcar/control/steering") {
-            Serial.println(message);
+            Serial.println(message);//Liam are these needed
```
Comment (Lindet94) [inline arduino/smartcar/smartcar.ino]: Same as above.

File: arduino/smartcar/smartcar.ino [hunk 4]
```diff
@@ -189,12 +199,75 @@ void loop()
...
+        {
+            gyroscopeAngle = gyroscope.getHeading();
+            angleAfterUpdate = gyroscopeAngle;
+            
+        }
+        else if (loopControl == 7)
+        {
+            carSpeed = car.getSpeed();
+
+            checkAngleDegree = angleAfterUpdate - angleBeforeUpdate;//checks if angle of change past the loop
```
Comment (Lindet94) [inline arduino/smartcar/smartcar.ino:216]: Don't understand the comment here.

File: arduino/smartcar/smartcar.ino [hunk 5]
```diff
@@ -11,21 +11,38 @@
...
+
+int angleChangeDegree;
+int saveAngleOfChange;
+int checkAngleDegree;
+
+bool safeStationaryAngleChange = true;
+
+double carSpeed;
+
+//controls which sensors active during the loops for the simulator car
```
Comment (Lindet94) [inline arduino/smartcar/smartcar.ino]: Check spelling in comment

File: arduino/smartcar/smartcar.ino [hunk 6]
```diff
@@ -11,21 +11,38 @@
...
-// incomingAvoidanceThreshold
+//incomingAvoidanceThreshold
 bool activeAvoidance = false;
 
-// controls which sensors active during the loops for the simulator car
+//registerCollision
+bool collision = false;
+int timeForCollision = 0;
+
+int angleBeforeUpdate;//needed for figure out angle change degree
```
Comment (Lindet94) [inline arduino/smartcar/smartcar.ino:34]: Check spelling in comment

File: arduino/smartcar/smartcar.ino [hunk 7]
```diff
@@ -359,10 +432,37 @@ void incomingAvoidanceThreshold(long frontUltDis, long backUltDis)
...
+        collision = false;
+    }
+    
+}
+
 void setDriveBackwards(bool value) // Added these to make the above if statement less repetitive
 {
     const auto currentTime = millis();
-    driveBackwards = value;
+    drivebackwards = value;
```
Comment (Lindet94) [inline arduino/smartcar/smartcar.ino]: Should use camelcase in variable

**Review by Lindet94 — CHANGES_REQUESTED** [review]
Code looks code as far as I can see. There is just some comments that needs to be clarified and checked for spelling errors. Found a variable that is not written in camel case as well. See comments :)

File: arduino/smartcar/smartcar.ino (hunk 2 above)
Comment (liamaxelrod) [reply arduino/smartcar/smartcar.ino]: I was asking if the Serial.println(message); was needed or if I can remove it for whoever reviewed my code.
Also I thought I got rid of that other comment so I'll get rid of it

File: arduino/smartcar/smartcar.ino (hunk 3 above)
Comment (liamaxelrod) [reply arduino/smartcar/smartcar.ino]: Same as before is it needed the Serial.println(message);

File: arduino/smartcar/smartcar.ino (hunk 4 above)
Comment (liamaxelrod) [reply arduino/smartcar/smartcar.ino:216]: Pretty much there is a problem every time the loop goes from 360 to 1 so what this does is it checks if it hit that mark and if it does it does not give any input for those couple of milliseconds

File: arduino/smartcar/smartcar.ino (hunk 7 above)
Comment (liamaxelrod) [reply arduino/smartcar/smartcar.ino]: Don't know what exactly you mean

**Comment by Lindet94** [comment]
Change title it something a bit more meaningful and also clarify the explanation in the pull request.

File: arduino/smartcar/smartcar.ino (hunk 2 above)
Comment (liamaxelrod) [reply arduino/smartcar/smartcar.ino]: Has been removed

File: arduino/smartcar/smartcar.ino (hunk 5 above)
Comment (liamaxelrod) [reply arduino/smartcar/smartcar.ino]: I changed it to say simulated car first in the comment

**Review by Lindet94 — APPROVED** [review]

**Pull request merged**